### PR TITLE
newVote function completed

### PR DESCRIPTION
**Pull request by RichardNeat**
Ticket 5. newVote completed to increment the value of "votes" on the reviews table. Error handling also added for invalid review_id, valid but non existent review_id, invalid inc_votes value and inv_votes key error.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -69,3 +69,50 @@ describe('GET /api/reviews/:review_id', () => {
...
+            review_body: 'Farmyard fun!',
+            review_img_url: 'https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png',
+            created_at: "2021-01-18T10:00:20.514Z",
+            votes: 3
+          }
+        return request(app).patch('/api/reviews/1').send(input).expect(201)
+        .then(({body}) => {
+            expect(body.review).toEqual(expected);
+        });
+    })
```
Comment (aswain03) [inline __tests__/app.test.js:91]: if we wanted to be fun we could also test for decreasing votes too!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -69,3 +69,50 @@ describe('GET /api/reviews/:review_id', () => {
             });
     });
 });
+
+describe('PATCH /api/reviews/:review_id', () => {
+    test('responds with status: 201 and and with the updated review incremeneted or decremented by the passed amount', () => {
```
Comment (aswain03) [inline __tests__/app.test.js:74]: 201 is created. 
are we creating anything here?

File: app.js [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
 
+app.patch('/api/reviews/:review_id', newVote);
+
 app.all('*', (req, res) => {
     res.status(404).send({msg: 'bad path'})
     });
 
 ///////////////////////////////////////////
-// EHMF's
+// EHMFs
```
Comment (aswain03) [inline app.js:22]: EHMFs are getting a bit big in the app now, have a look at putting them in their own file

**Comment by aswain03** [comment]
flying today! 🪰 

it would be worth splitting your controller and model files into there respective folders for each use. 
e.g. categories  will have a categories.controllers file and a categories.models.

File: __tests__/app.test.js (hunk 1 above)
Comment (RichardNeat) [reply __tests__/app.test.js:91]: Done

**Pull request merged**